web/src/main.jsx

```diff
@@ -11,26 +11,119 @@ import { AppointmentPage } from "./pages/AppointmentPage.jsx";
 import { WipPage } from "./pages/WipPage.jsx";
 import { MessagePage } from "./pages/MessagePage.jsx";
 import { ServiceHistoryPage } from "./pages/ServiceHistoryPage.jsx";
+import {
+  ClerkProvider,
+  SignedIn,
+  SignedOut,
+  RedirectToSignIn,
+  SignIn,
+  SignUp,
+  UserButton,
+} from "@clerk/clerk-react";
+
+if (!import.meta.env.VITE_CLERK_PUBLISHABLE_KEY) {
+  throw new Error("Missing Publishable Key");
+}
+const clerkPubKey = import.meta.env.VITE_CLERK_PUBLISHABLE_KEY;
+
 const defaultTheme = createTheme();
 const router = createBrowserRouter([
   {
     path: "/",
     element: <RootLayout />,
     children: [
       { index: true, element: <LoginPage /> },
-      { path: "/welcome", element: <WelcomePage /> },
-      { path: "/appointment", element: <AppointmentPage /> },
-      { path: "/wip", element: <WipPage /> },
-      { path: "/message", element: <MessagePage /> },
-      { path: "/history", element: <ServiceHistoryPage /> },
+      {
+        path: "/sign-in/*",
+        element: <SignIn routing="path" path="/sign-in" />,
+      },
+      {
+        path: "/sign-up/*",
+        element: <SignUp routing="path" path="/sign-up" />,
+      },
+      {
+        path: "/welcome",
+        element: (
+          <>
+            <SignedIn>
+              <WelcomePage />
+              <UserButton afterSignOutUrl="http://localhost:5173/" />
+            </SignedIn>
+            <SignedOut>
+              <RedirectToSignIn />
+            </SignedOut>
+          </>
+        ),
+      },
+      {
+        path: "/appointment",
+        element: (
+          <>
+            <SignedIn>
+              <AppointmentPage />
+              <UserButton afterSignOutUrl="http://localhost:5173/" />
+            </SignedIn>
+            <SignedOut>
+              <RedirectToSignIn />
+            </SignedOut>
+          </>
+        ),
+      },
+      {
+        path: "/wip",
+        element: (
+          <>
+            <SignedIn>
+              <WipPage />
+              <UserButton afterSignOutUrl="http://localhost:5173/" />
+            </SignedIn>
+            <SignedOut>
+              <RedirectToSignIn />
+            </SignedOut>
+          </>
+        ),
+      },
+      {
+        path: "/message",
+        element: (
+          <>
+            <SignedIn>
+              <MessagePage />
+              <UserButton afterSignOutUrl="http://localhost:5173/" />
```

Comment:
You will have to make this URL dynamic. When you deploy your code to `xyz.com`, `localhost:5173` will not be available. As this affects everyone, I will prepare a solution for that which I can share with the whole team.